src/App.tsx

```diff
@@ -1,74 +1,132 @@
-import React from 'react';
+import React, { useState, useMemo, useCallback, useRef } from 'react';
+import classNames from 'classnames';
+import debounce from 'lodash.debounce';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [delay, setDelay] = useState(300);
+  const applyQuery = useCallback(debounce(setQuery, delay), []);
```

Comment:
The `applyQuery` function is using `debounce` with an empty dependency array. This means it will not update when `delay` changes. Consider adding `delay` to the dependency array to ensure the debounce delay updates correctly when `delay` changes.